### PR TITLE
update shared creds/config and waiter documentation

### PR DESCRIPTION
**Pull request by skotambkar**
* [x] Adds documentation update for shared config and credentials file behavior in the SDK.
* [x] Adds waiter documentation and examples.

### CODE REVIEW

File: content/en/docs/making-requests.md [hunk 1]
```diff
@@ -426,6 +426,195 @@ functional arguments to `NextPage`. For more information about overriding client
...
+and error-prone. This is especially true when you are dealing with multiple 
+AWS resources, as different resources often have different polling APIs and 
+success states that indicate the availability of that particular resource
+
+The waiters help take the complexity out of it and are simple APIs that handle 
+the polling task for you. It also provides polling configurations, such as the 
+maximum wait time, minimum delay and maximum delay between polling retries. 
+Waiters use a back-off strategy between each attempt. With the waiters utility, 
+you can focus on working with the resource without worrying how to make sure 
+that this resource is ready.
```
Comment (skmcgrail) [inline content/en/docs/making-requests.md:448]: This section seems a bit wordy and bit of an unnecessary lengthy of an explanation.

File: content/en/docs/making-requests.md [hunk 2]
```diff
@@ -426,6 +426,195 @@ functional arguments to `NextPage`. For more information about overriding client
...
+
+The following example shows how you can use waiters to poll if a 
+{{% alias service=DDB %}} table is created and ready for a write operation.
```
Comment (skmcgrail) [inline content/en/docs/making-requests.md:451]: For example, you can use waiters to

File: content/en/docs/making-requests.md [hunk 3]
```diff
@@ -426,6 +426,195 @@ functional arguments to `NextPage`. For more information about overriding client
...
+    log.Printf("error: %v", err)
+    return 
+}
+fmt.Println("Dynamodb table is now ready for write operations")
+
+```
+
+Since the waiter constructor takes in a client interface, you can 
+provide a custom client that will be used by the waiter to poll the resource.
+This design also makes it easier to mock the waiter for testing purposes,
```
Comment (skmcgrail) [inline content/en/docs/making-requests.md:498]: I would drop this until we add a relevant section to the mocking / testing guide.

File: content/en/docs/making-requests.md [hunk 4]
```diff
@@ -426,6 +426,195 @@ functional arguments to `NextPage`. For more information about overriding client
...
+
+#### Overriding waiter configuration
+
+By default, SDK uses the minimum delay and maximum delay value configured with
```
Comment (skmcgrail) [inline content/en/docs/making-requests.md:502]: By default, the SDK

File: content/en/docs/making-requests.md [hunk 5]
```diff
@@ -426,6 +426,195 @@ functional arguments to `NextPage`. For more information about overriding client
...
+Since the waiter constructor takes in a client interface, you can 
+provide a custom client that will be used by the waiter to poll the resource.
+This design also makes it easier to mock the waiter for testing purposes,
+
+#### Overriding waiter configuration
+
+By default, SDK uses the minimum delay and maximum delay value configured with 
+optimal values defined by AWS services for different APIs. You can override these
+configurations on the entire waiter or per request, via waiter-specific 
+functional options present for each waiter.
```
Comment (skmcgrail) [inline content/en/docs/making-requests.md:505]: You can override waiter configuration by providing functional options during waiter construction, or when invoking a waiter operation.

File: content/en/docs/making-requests.md [hunk 6]
```diff
@@ -426,6 +426,195 @@ functional arguments to `NextPage`. For more information about overriding client
...
+This design also makes it easier to mock the waiter for testing purposes,
+
+#### Overriding waiter configuration
+
+By default, SDK uses the minimum delay and maximum delay value configured with 
+optimal values defined by AWS services for different APIs. You can override these
+configurations on the entire waiter or per request, via waiter-specific 
+functional options present for each waiter. 
+
+Here's how you can configure waiter-level override configuration
```
Comment (skmcgrail) [inline content/en/docs/making-requests.md:507]: For example. to override the ... waiter configuration:

File: content/en/docs/making-requests.md [hunk 7]
```diff
@@ -426,6 +426,195 @@ functional arguments to `NextPage`. For more information about overriding client
...
+fmt.Println("Dynamodb table is now ready for write operations")
+
+```
+
+#### Advanced waiter configuration overrides
+
+In the new design of waiters, SDK provides advanced capabilities such as overriding how SDK determines, 
+if a waiter should retry based on the provided input or obtained api response. This functionality is 
+helpful for users who want to additionally customize and control the waiter behavior. The waiter-specific 
+options also provides `APIOptions` to [customize operation middlewares](https://aws.github.io/aws-sdk-go-v2/docs/middleware/#writing-a-custom-middleware).
```
Comment (skmcgrail) [inline content/en/docs/making-requests.md:576]: Since this is a configuration guide I wouldn't add language like "In the new design".

Maybe simplify it to something like: "You can override the waiter default behavior by providing a custom retryable function".

File: content/en/docs/making-requests.md [hunk 8]
```diff
@@ -426,6 +426,195 @@ functional arguments to `NextPage`. For more information about overriding client
...
+
+Here's how you can configure advanced waiter overrides.
```
Comment (skmcgrail) [inline content/en/docs/making-requests.md:578]: For example, to...

File: content/en/docs/configuring-sdk/_index.md [hunk 9]
```diff
@@ -183,6 +218,36 @@ aws_access_key_id = <YOUR_TEMP_ACCESS_KEY_ID>
 aws_secret_access_key = <YOUR_TEMP_SECRET_ACCESS_KEY>
 aws_session_token = <YOUR_SESSION_TOKEN>
 ```
+The section name for a non-default profile within a credentials file 
+must not begin with the word `profile`. You can read more at 
+[AWS Tools and SDKs Shared Configuration and Credentials Reference Guide](https://docs.aws.amazon.com/credref/latest/refdocs/file-format.html#file-format-creds).
+
+#### Creating the Config File
+
+If you don't have a shared credentials file (`.aws/config`), you
```
[the remark this answers is not in the document]
Comment (skotambkar) [reply content/en/docs/configuring-sdk/_index.md:227]: Want to avoid `~` as not valid for Windows.

**Review by skmcgrail — APPROVED** [review]

**Review by jasdel — APPROVED** [review]
